modelator/src/lib.rs

```diff
@@ -145,34 +183,41 @@ impl ModelatorRuntime {
         &self,
         tla_tests_file_path: P,
         tla_config_file_path: P,
-    ) -> Result<Vec<Result<artifact::JsonTrace, Error>>, Error> {
+    ) -> Result<BTreeMap<String, Result<Vec<artifact::JsonTrace>, Error>>, Error> {
+        // Each test maps to a result containing the vec of all it's traces.
+
+        let mut res: BTreeMap<String, Result<Vec<artifact::JsonTrace>, Error>> = BTreeMap::new();
```

Comment:
possibly you can do the same as before here. I see a nested loop. so call `.flatten()` for flat out nested iterators.

Reply:
I'm sorry I think I'm missing the nested loop here. What is it?